lib/ansible/modules/packaging/os/yum_repository.py

```diff
@@ -717,6 +720,13 @@ def main():
     module.params['name'] = module.params['description']
     del module.params['description']
 
+    # Change list type to string for baseurl and gpgkey
+    for list_param in ['baseurl', 'gpgkey']:
+        if (
+                list_param in module.params and
+                isinstance(module.params[list_param], list)):
+            module.params[list_param] = "\n".join(module.params[list_param])
```

Comment:
Then down here, you can just convert the list into a string.

It would be nice to indent the subsequent lines, too.  A simple 4 space indent is fine or you can get more complex if you want:
``` python
for list_param in ['baseurl', 'gpgkey']:
    module.params[list_param] = "\n    ".join(module.params[list_param])
```